mithril-aggregator/src/http_server/routes/artifact_routes/cardano_transaction.rs

```diff
@@ -0,0 +1,302 @@
+
+                Ok(reply::internal_server_error(err))
+            }
+        }
+    }
+
+    /// Get Artifact by signed entity id
+    pub async fn get_artifact_by_signed_entity_id(
+        signed_entity_id: String,
+        http_message_service: Arc<dyn MessageService>,
+    ) -> Result<impl warp::Reply, Infallible> {
+        debug!("⇄ HTTP SERVER: artifact/{signed_entity_id}");
+
+        match http_message_service
+            .get_cardano_transaction_message(&signed_entity_id)
+            .await
+        {
+            Ok(Some(message)) => Ok(reply::json(&message, StatusCode::OK)),
+            Ok(None) => {
+                warn!("get_cardano_transaction_details::not_found");
+                Ok(reply::empty(StatusCode::NOT_FOUND))
+            }
+            Err(err) => {
+                warn!("get_cardano_transaction_details::error"; "error" => ?err);
+                Ok(reply::internal_server_error(err))
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+pub mod tests {
+    use crate::{
+        http_server::SERVER_BASE_PATH,
+        initialize_dependencies,
+        message_adapters::{
+            ToCardanoTransactionListMessageAdapter, ToCardanoTransactionMessageAdapter,
+        },
+        services::MockMessageService,
+    };
+    use chrono::{DateTime, Utc};
+    use mithril_common::{
+        entities::{Beacon, SignedEntity, SignedEntityType},
+        messages::ToMessageAdapter,
+        signable_builder::Artifact,
+        sqlite::HydrationError,
+        test_utils::{apispec::APISpec, fake_data},
+    };
+    use serde_json::Value::Null;
+    use warp::{http::Method, test::request};
+
+    use super::*;
+
+    fn setup_router(
+        dependency_manager: Arc<DependencyContainer>,
+    ) -> impl Filter<Extract = (impl warp::Reply,), Error = warp::Rejection> + Clone {
+        let cors = warp::cors()
+            .allow_any_origin()
+            .allow_headers(vec!["content-type"])
+            .allow_methods(vec![Method::GET, Method::POST, Method::OPTIONS]);
+
+        warp::any()
+            .and(warp::path(SERVER_BASE_PATH))
+            .and(routes(dependency_manager).with(cors))
+    }
+
+    pub fn create_signed_entities<T>(
+        signed_entity_type: SignedEntityType,
+        records: Vec<T>,
+    ) -> Vec<SignedEntity<T>>
+    where
+        T: Artifact,
+    {
+        records
+            .into_iter()
+            .enumerate()
+            .map(|(idx, record)| SignedEntity {
+                signed_entity_id: format!("{idx}"),
+                signed_entity_type: signed_entity_type.to_owned(),
+                certificate_id: format!("certificate-{idx}"),
+                artifact: record,
+                created_at: DateTime::parse_from_rfc3339("2023-01-19T13:43:05.618857482Z")
+                    .unwrap()
+                    .with_timezone(&Utc),
+            })
+            .collect()
+    }
+
+    #[tokio::test]
+    async fn test_cardano_transactions_get_ok() {
```

Comment:
@sfauvel made an interesting remark about this way of testing (used in other routes tests too).
We are not testing the HTTP code result of the request response here.
Which means that even if the query fails (with a code 500), the test would pass because `verify_conformity` just check that the response is valid.

Reply:
Yes, it's completely true. We should fix all the tests in a different PR.
I'll create an issue for that :+1: